api/auth.py

```diff
@@ -66,3 +70,35 @@ def wrapper(request, submit_id: int, *args, **kwargs):
         return function(request, submit_id, *args, **kwargs)
 
     return wrapper
+
+
+def require_submit_access(function):
```

Comment:
Actually, this is too magical. Adding stuff onto `request` is very non-obvious. Let's just create a function that will return a submit and validate access, and use that in the API endpoints, something like this:
```python
submit = get_submit_write_access(assignment_id, login, submit_num)
```